client/my-sites/upgrades/domain-search/free-trial-notice.jsx

```diff
@@ -2,34 +2,39 @@
  * External dependencies
  */
 import React from 'react';
+import { connect } from 'react-redux';
+import page from 'page';
 import i18n from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import { addCurrentPlanToCartAndRedirect, getCurrentPlan, getDayOfTrial } from 'lib/plans';
+import { cartItems } from 'lib/cart-values';
+import { getDayOfTrial } from 'lib/plans';
+import { addItem } from 'lib/upgrades/actions';
 import Notice from 'components/notice';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+import { getCurrentPlan, isRequestingSitePlans } from 'state/sites/plans/selectors';
 
 const FreeTrialNotice = React.createClass( {
 	propTypes: {
 		cart: React.PropTypes.object.isRequired,
-		sitePlans: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.oneOfType( [
-			React.PropTypes.object,
-			React.PropTypes.bool
-		] ).isRequired
+		// connected props
+		currentPlan: React.PropTypes.object,
+		isRequestingSitePlans: React.PropTypes.bool,
+		selectedSiteSlug: React.PropTypes.string
 	},
 
 	addPlanAndRedirect( event ) {
 		event.preventDefault();
-
-		addCurrentPlanToCartAndRedirect( this.props.sitePlans, this.props.selectedSite );
+		addItem( cartItems.planItem( this.props.currentPlan.productSlug ) );
+		page( `/checkout/${ this.props.selectedSiteSlug }` );
 	},
 
 	render() {
-		const { cart, sitePlans } = this.props,
-			isDataLoading = ! cart.hasLoadedFromServer || ! sitePlans.hasLoadedFromServer,
-			currentPlan = getCurrentPlan( sitePlans.data );
+		const { cart, currentPlan, isRequestingSitePlans: isRequestingPlans } = this.props,
+			isDataLoading = isRequestingPlans || ! cart.hasLoadedFromServer;
 
 		if ( isDataLoading || ! currentPlan.freeTrial ) {
```

Comment:
We'll likely need a to add a `	<QuerySitePlans siteId={ siteId } />` note that this should be rendered before any loading checks do an early return.